### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by philchh**
1. Put in a couple comments in the source files because I would like an explanation.
    ex: effect.js line 445  super.destroy(releaseChildren); //Effect.$superp.Destroy.call(this, releaseChildren);

2. Package can transpile individual files and create a bundle as well.

3. Brought in 'pure-min.css', 'side-menu.css', 'ui.js' and 'extern'(folder)  in the 'dist' folder. Could not resolve the paths for some reason. Might be my webpack config.

4. The webpack/babel setup is probably not the best.

### CODE REVIEW

File: src/Effect.js [hunk 1]
```diff
@@ -267,691 +273,694 @@ var Effect = Class(Entity, {
...
+        p.reset();
+        this._particleManager.releaseParticle(p);
+        this.removeInUse(i, p);
       }
       this._inUse[i] = [];
     }
 
-    Effect.$superp.Destroy.call(this, releaseChildren);
-  },
+    super.destroy(releaseChildren); //Effect.$superp.Destroy.call(this, releaseChildren);
```
Comment (leshow) [inline src/Effect.js:445]: Here?

File: src/ParticleManager.js [hunk 2]
```diff
@@ -2,11 +2,12 @@ import { fmod, lerp } from "./Utils";
 import { drawSprite } from "../exp/pixi_main";
 import EffectsLibrary from "./EffectsLibrary";
 
+const c_particleLimit = 5000;
```
Comment (leshow) [inline src/ParticleManager.js]: Move into ParticleManager

File: src/Utils.js [hunk 3]
```diff
@@ -111,7 +111,7 @@ export function lerp(a, b, fract) {
 export const g_randomSeed = 17;
 
 export function randomUnit() {
-  var x = Math.sin(g_randomSeed++) * 10000;
+  let x = Math.sin(this.g_randomSeed++) * 10000;
```
Comment (leshow) [inline src/Utils.js]: You don't have access to `this` here

File: src/Utils.js [hunk 4]
```diff
@@ -127,7 +127,7 @@ export function getDistance2D(fromx, fromy, tox, toy) /* ,false=false */ {
   w = tox - fromx;
   h = toy - fromy;
 
-  // if (GetDefaultArg(fast, false)) return w * w + h * h;
+  if (GetDefaultArg(fast, false)) return w * w + h * h;
```
Comment (leshow) [inline src/Utils.js]: getdefaultarg doesn't exist anymore, you need to write this without using it

File: src/Utils.js (hunk 3 above)
Comment (leshow) [reply src/Utils.js]: should probably use 'window' it looks like it's supposed to be a global variable

Comment (leshow) [reply src/Utils.js]: and not `const`

File: src/EffectsLibrary.js [hunk 5]
```diff
@@ -1,218 +1,204 @@
...
   // $singleton: true, // or is it
   static instance;
 
-    static c_particleLimit = 5000;
+  static c_particleLimit = 5000;
 
-    static globalPercentMin        = 0;
-    static globalPercentMax        = 20.0;
-    static globalPercentSteps      = 100.0;
+  static globalPercentMin = 0;
```
Comment (leshow) [inline src/EffectsLibrary.js:7]: FYI anything declared w/ `static` can't be called with `this.`

File: src/Effect.js (hunk 1 above)
Comment (philchh) [reply src/Effect.js:445]: You've explained this to me... thanks

Comment (philchh) [reply src/Effect.js:445]: Inline comment has been removed

File: src/Utils.js (hunk 4 above)
Comment (philchh) [reply src/Utils.js]: The original getDistance2D function has 5 args
function GetDistance2D( fromx, fromy, tox, toy, fast /*= false*/ )

Not sure how to handle this

File: src/ParticleManager.js (hunk 2 above)
Comment (philchh) [reply src/ParticleManager.js]: Created a static var

File: src/EffectsLibrary.js (hunk 5 above)
Comment (philchh) [reply src/EffectsLibrary.js:7]: No need to change anything here?

File: src/Utils.js (hunk 4 above)
Comment (philchh) [reply src/Utils.js]: Not sure what I've done here is good. If an arg is omitted does 'undefined' need to be set as a placeholder?

File: src/Utils.js (hunk 3 above)
Comment (philchh) [reply src/Utils.js]: Did create a const. Pretty sure the default val for g_randomSeed is 17.  Not sure what the return val of randomUnit is supposed to do.

Comment (philchh) [reply src/Utils.js]: g_randomSeed is only used once in Utils.js

**Pull request merged**